settings.php

```diff
@@ -58,14 +58,34 @@
         new moodle_url('/mod/customcert/download_all_certificates.php'), ''));
 }
 
+
+$settings->add(new admin_setting_heading('scheduledtaskconfig',
+    get_string('scheduledtaskconfigheading', 'customcert'),
+    get_string('scheduledtaskconfigdesc', 'customcert')));
+
+$settings->add(new admin_setting_configtext('customcert/certificatesperrun',
+    get_string('certificatesperrun', 'customcert'),
+    get_string('certificatesperrun_desc', 'customcert'),
+    0, PARAM_INT));
+$settings->add(new admin_setting_configcheckbox('customcert/includeinnotvisiblecourses',
+    get_string('includeinnotvisiblecourses', 'customcert'),
+    get_string('includeinnotvisiblecourses_desc', 'customcert'), 0));
+$settings->add(
```

Comment:
I am not convinced this setting is needed. It's confusing. What is a certificate end date? It's just randomly the course end date which isn't clear. Also, a course may end but still want to send out certificates, eg. it ends after the final exam, then its closed, teacher does grading and then the user gets the certificate - valid scenario. I say we should remove this setting completely. Thoughts?

Reply:
Hi Mark,

I apologize for the delay in responding as I was on an extended leave for nearly two months and did not review emails during that time.

Regarding the hidden categories and courses, I configured them to allow administrators to tailor settings according to their organization's needs. Different organizations have varied operational methods. For instance, some universities have between 200K to 500K enrollments per term and retain courses on Moodle for 3 to 10 years. These older courses are usually hidden and are rarely accessed for certificate issuance. Therefore, it might not be efficient to include these in certificate checks for the entire 10 years. Allowing administrators to manage this through configurations ensures flexibility and efficiency.

Similarly, for old courses, I provided options for administrators to decide the time frame for system checks. For example, the system could be set to check only the courses from the last six months. If there is a need to process older courses, administrators can adjust the configurations as necessary.

I hope this addresses your concerns regarding the settings. If there are any other issues you are referring to, please let me know.

Best regards,

Reply:
Thanks @mohamedmohamedatia. Makes sense.